routing/route_weight.cpp

```diff
@@ -2,27 +2,70 @@
 
 #include "routing/cross_mwm_connector.hpp"
 
+#include <type_traits>
+
 using namespace std;
 
+namespace
+{
+template <typename Number,
+          typename EnableIf = typename enable_if<is_integral<Number>::value, void>::type>
+bool SumWillOverflow(Number lhs, Number rhs)
+{
+  if (lhs > 0)
+    return rhs > numeric_limits<Number>::max() - lhs;
+  if (lhs < 0)
+    return rhs < numeric_limits<Number>::min() - lhs;
+  return false;
+}
+}  // namespace
+
 namespace routing
 {
 double RouteWeight::ToCrossMwmWeight() const
 {
-  if (m_nonPassThroughCross > 0)
+  if (m_nonPassThroughCross > 0 || m_accessCross > 0)
     return connector::kNoRoute;
   return GetWeight();
 }
 
+RouteWeight RouteWeight::operator+(RouteWeight const & rhs) const
+{
+  ASSERT(!SumWillOverflow(m_nonPassThroughCross, rhs.m_nonPassThroughCross), ());
```

Comment:
Could you please add values to the diagnostic message, here and below?

Reply:
done